test/hooks.test.js

```diff
@@ -145,10 +145,12 @@ test('onRequest hook should support encapsulation / 2', t => {
 })
 
 test('onRequest hook should support encapsulation / 3', t => {
-  t.plan(13)
+  t.plan(17)
   const fastify = Fastify()
+  fastify.decorate('hello', 'world')
 
-  fastify.addHook('onRequest', (req, res, next) => {
+  fastify.addHook('onRequest', function (req, res, next) {
```

Comment:
Can you also add a decorator *after* the hook registration?

Reply:
What's the reason?

Reply:
To be 100% sure that we are not missing the context reference :)

Reply:
I mean what does it verify. I don't get it. Can you make an example?

Reply:
Unlike the routes the hooks are registered synchronously.
We had some troubles with the async declaration of the routes, so I want to add a test now for the hooks, in this way if in the future we make thew hook api async, we can detect very quickly weird behaviours.

Reply:
Check